perun-core/src/main/java/cz/metacentrum/perun/core/bl/AttributesManagerBl.java

```diff
@@ -156,6 +156,24 @@
 	List<Attribute> getAttributes(PerunSession sess, Resource resource, Member member, boolean workWithUserAttributes) throws InternalErrorException, WrongAttributeAssignmentException;
 
 	/**
+	 * Gets selected <b>non-empty</b> attributes associated with the member on the resource and if workWithUserAttributes is
+	 * true, gets also all <b>non-empty</b> user, user-facility and member attributes.
+	 * Attributes are selected by list of attr_names. Empty list means all attributes.
+	 *
+	 * @param sess perun session
+	 * @param resource to get the attributes from
+	 * @param member to get the attributes from
+	 * @param workWithUserAttributes if true returns also user-facility, user and member attributes (user is automatically get from member a facility is get from resource)
+	 * @return list of selected attributes
+	 *
+	 * @throws InternalErrorException if an exception raise in concrete implementation, the exception is wrapped in InternalErrorException
+	 * @throws WrongAttributeAssignmentException
+	 *
+	 * !!WARNING THIS IS VERY TIME-CONSUMING METHOD. DON'T USE IT IN BATCH!!
+	 */
+	List<Attribute> getAttributes(PerunSession sess, Resource resource, Member member, List<String> attrNames, boolean workWithUserAttributes) throws InternalErrorException, WrongAttributeAssignmentException;
```

Comment:
This javadoc doesn't match method implementation or is slightly misleading at start. You mention attributes associated with member on resource, but doesn't specify exact namespace. Then you say ``workWithUserAttributes==true`` return in addition also user, user-facility and *member* attributes.

Method implementation *always returns member* and member-resource attributes, then if ``workWithUserAttributes==true`` it returns also user and user-facility attributes.

I'm not sure, what is our standard for attribute methods with member and resource params, but you should return always member-resource, then most probably also member (this i'm not sure) and if flag is true return also user and user-facility attributes.

Same goes for Entry layer javadoc.